Sources/AsyncHTTPClient/FileDownloadDelegate.swift

```diff
@@ -19,13 +19,36 @@ import NIOPosix
 /// Handles a streaming download to a given file path, allowing headers and progress to be reported.
 public final class FileDownloadDelegate: HTTPClientResponseDelegate {
     /// The response type for this delegate: the total count of bytes as reported by the response
-    /// "Content-Length" header (if available) and the count of bytes downloaded.
+    /// "Content-Length" header (if available), the count of bytes downloaded, and the
+    /// response head.
     public struct Progress: Sendable {
         public var totalBytes: Int?
         public var receivedBytes: Int
+
+        public var head: HTTPResponseHead {
+            get {
+                #if DEBUG
```

Comment:
I think it's probably easier to just write this as `assert(self._head != nil)`. That keeps the code common always, and lets us leverage assertions mode instead of relying on specifically having `DEBUG` defined.

Reply:
Also nit: we require explicit `self`.